benchmarks/squad/squad.py

```diff
@@ -0,0 +1,249 @@
+import re
+import string
+
+from typing import List, Tuple, Union
+
+import numpy as np
+from scipy.optimize import linear_sum_assignment  # type: ignore
+
+from inspect_ai import Task, task
+from inspect_ai.dataset import Sample, hf_dataset
+from inspect_ai.scorer import Score, Target, bootstrap_std, mean, scorer
+from inspect_ai.solver import system_message, generate, TaskState
+
+
+SYSTEM_MESSAGE = """
+Using only the context provided in the question itself, you are tasked with answering a question in as few words as possible.
+"""
+
+@task
+def squad():
+   
+    dataset = hf_dataset(
+        path="rajpurkar/squad",
+        split="validation",
+        sample_fields=record_to_sample,
+        trust=True,
+        shuffle=True
+    )
+
+    return Task(
+        dataset=dataset,
+        plan=build_plan(),
+        scorer=squad_f1_scorer()
+    )
+
+def record_to_sample(record):
+  return Sample(
+      input=format_input(record),
+      target=format_target(record),
+      id=record["id"],
+  )
+
+def format_input(record) -> str:
+  passage_str = f"""Context: {record["context"]}"""
+  question_str = f"""Question: {record["question"]}"""
+  input_str = "\n".join([passage_str, question_str])
+
+  return input_str
+
+
+def format_target(record) -> List[str]:
+  answers = record["answers"]
+  choices = answers["text"]
+
+  target = choices
+  return target
+
+def build_plan() -> List:
+    
+  sys_msg = system_message(SYSTEM_MESSAGE)
+
+  plan = [
+    sys_msg,
+    generate(),
+  ]
+
+  return plan
+
+
+@scorer(metrics=[mean(), bootstrap_std()])
+def squad_f1_scorer():
+    async def score(state: TaskState, target: Target) -> Score:
+        # Get generated answer and extract relevant answer text
+        prediction = state.output.completion
+        ref_answers = target.target
+
+        # Compute exact match (EM) and F1 score
+        em_score, f1_score = await process_results(prediction, ref_answers)
+
+        # F1 score and EM reported as main aggregated metrics, EM score currently added to metadata
+        return Score(
+            value=f1_score,
+            answer=prediction,
+            metadata={
+                "metrics": {
+                    "f1_score": f1_score,
+                    "em_score": em_score,
+                },
+                "gold_answers": ref_answers,
+            },
+        )
+
+    return score
+
+# From here until '_normalize()' is copied from
```

Comment:
We've also done the same for the `drop` eval. Feels like maybe this should be something that we fold into Inspect itself? @dragonstyle Could you take a look at the scoring/metrics going on here and in drop (already merged) and think about what we should consider doing?

Reply:
Agreed, it seems to be a common choice for scoring/metrics amongst a few benchmarks - and worth noting this logic is almost identical to the drop logic (bar some variable names).

Reply:
Note that we've got a PR here (https://github.com/UKGovernmentBEIS/inspect_ai/tree/feature/f1) to add `f1()` and `exact()` scorers. Once that is merged you should be able to use them rather than include all of this extra scoring code.